scripts/zones/Waughroon_Shrine/mobs/Princess_Jelly.lua

```diff
@@ -0,0 +1,172 @@
+-----------------------------------
+-- Area: Waughroon Shrine
+--  MOB: princess_jelly
+-- Author: Setzor (ported to LSB by dallano)
+-----------------------------------
+local ID = require("scripts/zones/Waughroon_Shrine/IDs")
+require("scripts/globals/pathfind")
+require("scripts/globals/utils")
+require("scripts/globals/spell_data")
+-----------------------------------
+local elementalSpells =
+{
+    {xi.magic.spell.BURN,  xi.magic.spell.FIRE},
+    {xi.magic.spell.DROWN, xi.magic.spell.WATER},
+    {xi.magic.spell.SHOCK, xi.magic.spell.THUNDER},
+    {xi.magic.spell.RASP , xi.magic.spell.STONE},
+    {xi.magic.spell.CHOKE, xi.magic.spell.AERO},
+    {xi.magic.spell.FROST, xi.magic.spell.BLIZZARD},
+    {xi.magic.spell.DIA,   xi.magic.spell.BANISH},
+    {xi.magic.spell.BIO,   xi.magic.spell.DRAIN},
+}
+local centers =
+{
+    {-177.5, 60,-142},
+    {  22.5,  0,  18},
+    { 222.5,-60, 138},
+}
+
+local entity = {}
+
+entity.onMobInitialize = function(mob)
+    mob:setMobMod(xi.mobMod.MAGIC_COOL, 40)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:setSpeed(2)
+    mob:setUnkillable(false)
+    mob:setLocalVar('mobElement', math.random(1,8))
+    local element = mob:getLocalVar('mobElement')
+
+    if element == 1 then
```

Comment:
Could use a keyed table instead of all this `elseif` conditionaling.

```lua
local element = mob:getLocalVar('mobElement')
local mevaList =
{
    [0] = xi.mod.LIGHT_MEVA,
    [1] = xi.mod.WATER_MEVA,
    [2] = xi.mod.THUNDER_MEVA,
    [3] = xi.mod.EARTH_MEVA,
    [4] = xi.mod.WIND_MEVA,
    [5] = xi.mod.ICE_MEVA,
    [6] = xi.mod.FIRE_MEVA,
    [7] = xi.mod.DARK_MEVA
}
mob:addMod(mevaList[element], -250)
```
A variable not set with always return zero, so I just put light at the top as zero instead of an `else`.